### PR TITLE
feat: enable selective query nesting for bulk data processing

### PR DESCRIPTION
**Pull request by MicBun**
Review after: https://github.com/trufnetwork/kwil-db/pull/1587

Implements selective relaxation of ErrQueryActive restriction to enable
efficient bulk operations while maintaining database safety and security.

### Impact
- Enables performance improvement for bulk data operations
- Reduces resource consumption through efficient SQL execution
- Supports standard SQL patterns like WITH RECURSIVE + DML
- Improves developer experience with intuitive SQL behavior

### Technical Changes
- Add query analysis module to detect safe vs unsafe nesting patterns
- Implement selective ErrQueryActive bypass for proven-safe operations:
  * WITH RECURSIVE combined with INSERT/UPDATE/DELETE operations
  * Simple DELETE/UPDATE queries with subqueries for bulk operations
- Maintain existing security restrictions for dangerous patterns:
  * Complex nested subqueries with excessive depth (>6 levels)
  * Nested EXISTS and IN clauses
  * Regular DML operations without legitimate bulk patterns

### Key Features
- Smart pattern detection distinguishes bulk operations from security risks
- Case-insensitive SQL keyword handling
- Comprehensive boundary and nesting depth validation
- Robust error handling for malformed queries and edge cases

### Files Modified
- node/engine/interpreter/context.go: Enhanced query execution logic
- node/engine/interpreter/query_analysis.go: New query analysis module
- node/engine/interpreter/query_analysis_test.go: Comprehensive test suite

### Test Coverage
- Positive, negative, and boundary condition testing
- Security vulnerability prevention validation
- Error condition and robustness testing

Resolves performance bottlenecks in bulk data processing operations
while preserving database security and maintaining backward compatibility
for existing query restrictions.

resolves: https://github.com/trufnetwork/kwil-db/issues/1588

## Summary by CodeRabbit

* **New Features**
  * Added table-valued function support with UNNEST in FROM/JOIN (requires table alias; optional column aliases; validates args; emits proper SQL).
  * Planner, planner nodes, AST, parser, and SQL generator now handle table-function relations and include them in plans and output.
  * Stateful nested-query handling with finer-grained safety checks and clearer errors for disallowed nesting.

* **Tests**
  * Comprehensive unit and integration tests for UNNEST parsing, planning, SQL generation, and nested-query safety.

### CODE REVIEW

File: node/engine/interpreter/query_analysis.go [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+// Precompiled regexes for efficient pattern matching
+var (
+	subqueryRe = regexp.MustCompile(`(?is)\(\s*SELECT\b`)
+
+	// Same-statement WITH RECURSIVE + DML detection (no intervening semicolon)
+	reWithRecursiveDMLSameStmt = regexp.MustCompile(`(?is)\bWITH\s+RECURSIVE\b[^;]*\b(INSERT|UPDATE|DELETE)\b`)
+
+	// Problematic nesting patterns - precompiled with case-insensitive and dotall flags
+	// These patterns detect genuinely dangerous constructs while avoiding false positives on WITH RECURSIVE
+	problematicNestRes = []*regexp.Regexp{
```
[the remark this answers is not in the document]
Comment (MicBun) [reply node/engine/interpreter/query_analysis.go:17]: > are any of these bad queries related to determinism?
No, those are for Performance issue. Not determinism. 

> So isn't it ok to just act as PostgreSQL would, and just be concerned about blocking nondeterminism, leaving performance issues to db owners?

Yes, we can do that. Should we remove the performance-based restrictions entirely and only block actual consensus-breaking patterns? It would behave more closely like PostgreSQL. I implement the safe pattern to respect the previous intention of blocking nested query.

@outerlook

**Review by outerlook — APPROVED** [review]
